genie-common/src/main/java/com/netflix/genie/common/util/GenieDateFormat.java

```diff
@@ -65,4 +77,27 @@ public StringBuffer format(final Date date, final StringBuffer toAppendTo, final
             )
         );
     }
+
+    /**
+     * JsonSerializer to output (Optional) Date using GenieDateFormat.
+     *
+     * @since 3.3.11
+     */
+    public static class OptionalDateJsonSerializer extends JsonSerializer<Optional<Date>> {
```

Comment:
I don't see you annotating any `Optional` dates as you have the annotation on members themselves. Does jackson call the getters instead and that's why you have it like this?

Reply:
Yes